### ISSUE TITLE
"dnx . kestrel" failing due to missing libuv on CentOS 7

### ISSUE THREAD

**Issue by ganeshran** [aspnet/Home#760]
I am able to execute Console apps but not the HelloMvC from the samples project. I tried the fix suggested of building libuv locally and then creating a symlink on /usr/lib/ and /usr/local/lib and /usr/lib64. dnx is still not able to load libuv. 

Also tried to install libuv from the Centos repo but it didnt work. 

dnu restore worked fine under mono runtime but after switching to coreclr to execute, this error is thrown.

```
System.InvalidOperationException: Unable to load libuv. Make sure libuv is installed and available as libuv.so.1
   at Microsoft.AspNet.Server.Kestrel.Networking.Libuv.Load(String dllToLoad)
   at Kestrel.ServerFactory.Start(IServerInformation serverInformation, Func`2 application)
   at Microsoft.AspNet.Hosting.Internal.HostingEngine.Start()
   at Microsoft.AspNet.Hosting.Program.Main(String[] args)
```

Output of dnvm list. I am running the latest samples.

```
Active Version              Runtime Arch Location             Alias
------ -------              ------- ---- --------             -----
       1.0.0-beta5          mono         ~/.dnx/runtimes      default
       1.0.0-beta5          coreclr x64  ~/.dnx/runtimes      
       1.0.0-beta6          coreclr x64  ~/.dnx/runtimes      
  *    1.0.0-beta7-12264    coreclr x64  ~/.dnx/runtimes      
       1.0.0-beta7-12264    mono         ~/.dnx/runtimes      
```

**Comment by Eilon** [comment]
@halter73

**Comment by ganeshran** [comment]
@Eilon @halter73  I figured out the issue. For some reason the dlopen call didn't detect the symlinks (on CentOS 7 ) from the built libuv to the /usr/lib and /usr/lib64 folder. kestrel starts after physically coping the libuv.so.1 file to the folder and running the console application.

Now kestrel runs but I get execption for other assemblies

```C#
        [DllImport("libdl")]
                public static extern IntPtr dlopen(String fileName, int flags);

        [DllImport("libdl")]
                public static extern IntPtr dlsym(IntPtr handle, String symbol);

        [DllImport("libdl")]
                public static extern int dlclose(IntPtr handle);

        [DllImport("libdl")]
                public static extern IntPtr dlerror();

        public Func<string, IntPtr> LoadLibraryNew;
        public static void Main()
        {
                new Program().Load();
                Console.WriteLine("Program over");
        }

        public Program()
        {
                this.LoadLibraryNew = LoadLibrary;
        }

        public void Load()
        {
                var module = dlopen("./libuv.so.1",2);
                if(module == IntPtr.Zero)
                {
                        Console.WriteLine("Module Not loaded");
                        var errPtr = dlerror();
                        throw new Exception("dlsym: " + Marshal.PtrToStringAnsi(errPtr));
                }
        }

**Comment by ganeshran** [comment]
Kestrel runs but I am getting a new exception for Runtime.Serialization.Primitives assembly. 

Should I create a new issue or continue in this one itself?

>error   : [Microsoft.AspNet.Diagnostics.ErrorPageMiddleware] An unhandled exception has occurred while executing the request
System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.Serialization.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

>File name: 'System.Runtime.Serialization.Primitives, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load the specified file.
>File name: 'System.Runtime.Serialization.Primitives'
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyName(AssemblyName assemblyName)
   at System.Runtime.Loader.AssemblyLoadContext.Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
   at Microsoft.AspNet.Mvc.MvcJsonMvcOptionsSetup.<>c__DisplayClass0_0.<.ctor>b__0(MvcOptions _)
   at Microsoft.Framework.OptionsModel.OptionsManager`1.<>c__DisplayClass5_0.<Configure>b__1(TOptions options, IConfigureOptions`1 setup)
   at System.Linq.Enumerable.Aggregate[TSource,TAccumulate](IEnumerable`1 source, TAccumulate seed, Func`3 func)
   at Microsoft.Framework.OptionsModel.OptionsManager`1.GetNamedOptions(String name)
   at Microsoft.AspNet.Mvc.ApplicationModels.DefaultApplicationModelProvider..ctor(IO

**Comment by ganeshran** [comment]
Output of dnvm list. I switched to coreclr before executing kestrel. 

I tried switching to mono as runtime and it works perfectly. But it doesn't work with coreclr runtime. Is this expected behaviour? 

My dot net verison is beta7-12264 and the sample being executed is the HelloMvc project in *latest* folder in this repo

>
Active Version              Runtime Arch Location             Alias
------ -------              ------- ---- --------             -----
       1.0.0-beta5          mono         ~/.dnx/runtimes      default
       1.0.0-beta5          coreclr x64  ~/.dnx/runtimes      
       1.0.0-beta6          coreclr x64  ~/.dnx/runtimes      
  *    1.0.0-beta7-12264    coreclr x64  ~/.dnx/runtimes      
       1.0.0-beta7-12264    mono         ~/.dnx/runtimes
